### ISSUE TITLE
Severe CPU performance regression on Oculus Quest (Android)

### ISSUE THREAD

**Issue by BHSPitMonkey** [Unity-Technologies/InputSystem#1109]
After updating from `preview.3` to `preview.4`, my application's framerate plummets to around 24fps when running on an Oculus Quest (about a third of what is normal/required). Based on some profiling, this seems to be caused by some very long execution times in the [InputCheckForUnpairedDeviceActivity](https://github.com/Unity-Technologies/InputSystem/blob/922e75387afc00277bfcf0267b5233d54b31c8d1/Packages/com.unity.inputsystem/InputSystem/Plugins/Users/InputUser.cs#L1691-L1750) sample from `InputUser`:

![image](https://user-images.githubusercontent.com/33672/77886882-7d993780-721e-11ea-956f-2ae83c2e6903.png)

I can reproduce this using `preview.4`, `preview.5`, and `preview.6`. Returning to `preview.3` mitigates the problem. I'm also using:

- XR Management 3.0.6
- XR Legacy Input Helpers 1.3.11
- Oculus XR Plugin 1.2.0

**Comment by Rene-Damm** [comment]
Could you file a ticket for this with the Unity bug reporter? Something we need to take a look at.

Are you using PlayerInput? If so, and in case you don't need automatic device switching (which I assume is the cause for this path getting enabled), turning off "Auto-Switch" on PlayerInputs should work around the problem for now.

**Comment by BHSPitMonkey** [comment]
Sure; I just created Case 1232227 in FogBugz with the same information.

I'm using a PlayerInput, and indeed disabling "Auto-Switch" does appear to negate the performance issue on previews 4 through 6 -- though at first glance it also seems to disrupt my ability to use controller input if the controllers are discovered _after_ the game starts (which is pretty typical for me if I "Build and Run", then put on the headset, then pick up controllers).

**Comment by stefanunity** [comment]
Hello,

we're going to close all the github issues and will concentrate bug information in the Unity public issue tracker https://issuetracker.unity3d.com/product/unity/issues?project=116&status=1&unity_version=&view=hottest.

This issue is currently with the Customer QA team and we're trying to reproduce internally.

Thank you!

**Comment by BHSPitMonkey** [comment]
For what it's worth, I this is still happening today on 1.0.0 (I decided to give the latest 2020.1 Editor beta a try, and it forced me to upgrade Input System from `preview.3` where I'd been holding off to avoid this bug). The conditions described in my original issue and comment above are still true (enabling Auto-Switch creates the same perf hit as before, and disabling it causes inputs not to work at all if the controllers were not already visible and being tracked when the game was started).

I spent a bit of time working with QA via FogBugz and even packaged up a new repro/demo project, but I think they gave up some time in May. I'm happy to share that project with anyone interested in seeing it.
